Modix/Modules/CatModule.cs

```diff
@@ -3,39 +3,51 @@
     using Discord.Commands;
     using Modix.Services.Cat;
     using System;
-    using System.Net.Http;
     using System.Threading;
     using System.Threading.Tasks;
+    using Serilog;
 
     public class CatModule : ModuleBase
     {
-        private static readonly HttpClient Client = new HttpClient();
+        private static Media MediaType;
+        private readonly ICatService _catService;
 
-        [Command("cat"), Summary("Gets a cat")]
-        public async Task Cat(string gif = "")
+        public CatModule(ICatService catService)
         {
-            // 5 seconds to find a cat
-            var cts = new CancellationTokenSource(5000);
-            var token = cts.Token;
+            _catService = catService;
+        }
 
-            var message = string.Empty;
+        public enum Media
+        {
+            Picture, // 0
+            Gif // 1
+        }
 
-            var cat = new CatService();
+        [Command("cat"), Summary("Gets a cat")]
+        public async Task Cat(string param = "")
+        {
+            var message = string.Empty;
 
-            // Regular picture
-            if (!string.Equals("gif", gif, StringComparison.OrdinalIgnoreCase))
+            try
             {
-                message = await cat.GetCatPicture(token);
-            }
-            else if (gif.Equals("gif", StringComparison.OrdinalIgnoreCase))
+                MediaType = (Media) Enum.Parse(typeof(Media), param, true);
+
+                if (!Enum.IsDefined(typeof(Media), MediaType) && !MediaType.ToString().Contains(","))
```

Comment:
if enum is not defined, and it doesn't contain a comma? An enum would never contain a comma...?